microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,514 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re, time
+
+from assertpy import assert_that
+
+from lisa import (
+    Node,
+    TestSuite,
+    TestSuiteMetadata,
+    TestCaseMetadata,
+    Environment
+)
+from lisa.base_tools.cat import Cat
+from lisa.operating_system import SLES, CentOs, Redhat
+from lisa.testsuite import TestCaseMetadata
+from lisa.tools.echo import Echo
+from lisa.util import SkippedException
+from lisa.util.logger import Logger
+
+def check_vm_feature_support_status(min_support_version: str, node: Node) -> bool:
+    if not min_support_version:
+        return False
+    
+    kernel_array = node.execute(
+        "uname -r | awk -F '[.-]' '{print $1,$2,$3,$4}'"
+    ).stdout.split()
+    support_kernel_array = re.split('\.|\-', min_support_version)
+    for indx in range(4):
+        if kernel_array[indx] > support_kernel_array[indx]:
+            return True
+        elif kernel_array[indx] < support_kernel_array[indx]:
+            return False
+    return True
+
+
+@TestSuiteMetadata(
+    area="power",
+    category="functional",
+    description="""
+    """
+)
+class power(TestSuite):
+    HB_CUSTOM_KERNEL_URL = "git://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git"
+    HB_CUSTOM_KERNEL_BRANCH = "stable"
+    HB_URL = ""
+
+    @TestCaseMetadata(
+        description="""
+        This test can be performed in Azure and Hyper-V both. But this script only covers Azure.
+        1. Prepare swap space for hibernation
+        2. Compile a new kernel (optional)
```

Comment:
Can hibernation work without this? If so, this step should be excluded in the test case.

Reply:
yes Lili told me we don't need to compile kernel in v3